click/core.py

```diff
@@ -1289,6 +1290,8 @@ def __init__(self, param_decls=None, type=None, required=False,
         self.is_eager = is_eager
         self.metavar = metavar
         self.envvar = envvar
+        if autocompletion is not None:
```

Comment:
Or maybe more explicitly:
`if cmd_param.autocompletion is not None`

Then the behavior is consistent if they incorrectly set autocompletion=false. Although that seems unlikely.